src/module/UserTermsCondition/user-terms-condition.controller.ts

```diff
@@ -0,0 +1,21 @@
+import { Body, Controller, Get, Param, Post } from "@nestjs/common";
+import { UserTermsConditionService } from "./user-terms-condition.services";
+
+@Controller('user-terms-condition')
+export class UserTermsConditionController {
+    constructor(private readonly userTermsConditionService: UserTermsConditionService) { }
+
+    @Post('/accept')
+    async acceptTermsCondition(
+        @Body('userId') userId: string,
+        @Body('userAgent') userAgent: string
+    ) {
+        return await this.userTermsConditionService.acceptTerms(userId, userAgent);
+    }
```

Comment:
Dont get userid from body Instead we can extract from jwt token 
using @User() decorator and also we can extract user ip address and user agent from @Req decorator

from req you can extract details like this

`req.ip,  req.get('User-Agent')`


```suggestion
    @Post('/accept')
    async acceptTermsCondition(
        @User("userId"). userId: string,
        @Req() req: Request
    ) {
        return await this.userTermsConditionService.acceptTerms(userId, req);
    }
```